src/api_server.h

```diff
@@ -44,6 +44,8 @@ class MetadataApiServer {
   ~MetadataApiServer();
 
  private:
+  friend class ApiServerTest;
+
```

Comment:
Another thing we can clean up here is making both `operator()` and `log()` `const` in `Dispatcher`. Can you please do this?